v2/tools/generator/internal/codegen/embeddedresources/remove_empty_objects.go

```diff
@@ -55,6 +55,11 @@ func findEmptyObjectTypes(
 			continue
 		}
 
+		// Don't prune top level status or spec types, they are allowed to be empty
+		if def.Name().IsStatus() || def.Name().IsSpec() {
```

Comment:
These methods only check based on naming convention - which is extended past the top level types in some cases.

Examples:
* In `apimanagement` - [BackendServiceFabricClusterProperties_STATUS](https://azure.github.io/azure-service-operator/reference/apimanagement/v1api20220801/#apimanagement.azure.com/v1api20220801.BackendServiceFabricClusterProperties_STATUS)
* In `cache` - [Persistence_STATUS](https://azure.github.io/azure-service-operator/reference/cache/v1api20210301/#cache.azure.com/v1api20210301.Persistence_STATUS)

You need another way to detect the top level spec/status.

Reply:
Good eye. Will fix!

Reply:
Fixed